### ISSUE TITLE
initialize()/postInitialize() order between components wrong on initially disabled entity

### ISSUE THREAD

**Issue by TobiasWehrum** [playcanvas/engine#1047]
I think I found a bug in the initialize()/postInitialize() order when later enabling an entity that was disabled when the app was started. To make matters worse, my app was still working with PlayCanvas 0.218.0 ded6028 and now isn't with 0.219.3 f5f75ce, so I assume it was introduced in the meantime. (I'm using the PlayCanvas editor so I can't roll back to check.)

When an entity has two scripts, I would assume the following on start:

1. Script A initialize()
2. Script B initialize()
3. Script A postInitialize()
4. Script B postInitialize()

And indeed that's what happens if the entity is enabled at start.

With PlayCanvas 0.219.3 f5f75ce, the following happens when the entity is disabled at start, but is enabled later:

1. Script A initialize()
2. Script A postInitialize()
3. Script B initialize()
4. Script B postInitialize()

I built a [minimal example project](https://playcanvas.com/editor/scene/569888) to demonstrate the problem. [Launch it](https://launch.playcanvas.com/569888?debug=true), open the console and press space to enable the second object.

**Comment by vkalpias** [comment]
Thanks this was introduced with this commit https://github.com/playcanvas/engine/commit/294a2ce6b9ff3f2116471fb5abb089f0bd648fcb which is attempting to fix a bug when individual scripts are enabled. We will have a look at fixing this as soon as possible.

**Comment by afk-mcz** [comment]
Hey Is this related to this issue ?

https://forum.playcanvas.com/t/entity-clone-not-deep-cloning-attributes/5374

**Comment by vkalpias** [comment]
I'm not sure but can you both check launching your application using this query parameter in the launch window URL:

`?use_local_engine=https://code.playcanvas.com/playcanvas-latest.js`

This will load the latest master branch where I've reverted that change. Let me know if this fixes your problems.

**Comment by afk-mcz** [comment]
Yes, this seems to fix the issue

**Comment by vkalpias** [comment]
@afk-mcz I would appreciate it if you could provide a test case for what your problem is exactly because I'm not sure I understand exactly what's broken in your situation. That's so we can fix it :) Thanks

**Comment by TobiasWehrum** [comment]
@vkalpias Yes, it fixes the problem I describe here.

**Comment by vkalpias** [comment]
I'm deploying a new version of the engine right now without the offending commit until we figure out what's wrong exactly.

**Comment by afk-mcz** [comment]
@vkalpias I tried to replicate it on a new blank project but I couldn't. I think I have a post Initialization script around that triggers the object initialization, and maybe it's getting called in the wrong order, and that's why my objects are not getting initialized.

I changed a lot of files trying to figure out the initial problem, so right now it's a little bit hard to reproduce in another project, will try later if the bug continues.
